pkg/openstack/nova.go

```diff
@@ -48,6 +49,111 @@ func ReconcileNova(ctx context.Context, instance *corev1beta1.OpenStackControlPl
 		return ctrl.Result{}, nil
 	}
 
+	// Create service overrides to pass into the service CR
+	// and expose the public endpoint using a route per default.
+	// Any trailing path will be added on the service-operator level.
+	apiServiceOverrides := map[string]service.OverrideSpec{}
+	serviceDetails := []ServiceDetails{}
+
+	for _, endpointType := range []service.Endpoint{service.EndpointPublic, service.EndpointInternal} {
+		sd := ServiceDetails{
+			ServiceName:         novav1.GetAPIServiceName(),
+			Namespace:           instance.Namespace,
+			Endpoint:            endpointType,
+			ServiceOverrideSpec: instance.Spec.Nova.Template.APIServiceTemplate.Override.Service,
+			RouteOverrideSpec:   instance.Spec.Nova.APIOverride.Route,
+		}
+
+		svcOverride, ctrlResult, err := sd.CreateRouteAndServiceOverride(ctx, instance, helper)
+		if err != nil {
+			return ctrlResult, err
+		} else if (ctrlResult != ctrl.Result{}) {
+			return ctrlResult, nil
+		}
+
+		serviceDetails = append(
+			serviceDetails,
+			sd,
+		)
+
+		apiServiceOverrides[string(endpointType)] = *svcOverride
+	}
+
+	// Create service overrides to pass into the service CR
+	// and expose the public endpoint using a route per default.
+	// Any trailing path will be added on the service-operator level.
+	metadataServiceOverrides := map[string]map[string]service.OverrideSpec{}
+	novncproxyServiceOverrides := map[string]map[string]service.OverrideSpec{}
+
+	// cell service override
+	for cellName, template := range instance.Spec.Nova.Template.CellTemplates {
+		cellOverride := instance.Spec.Nova.CellOverride[cellName]
+
+		// metadata
+		metadataServiceName := novav1.GetMetadataServiceName(&cellName)
+		metadataSD := ServiceDetails{
+			ServiceName:         metadataServiceName,
+			Namespace:           instance.Namespace,
+			Endpoint:            service.EndpointInternal,
+			ServiceOverrideSpec: template.MetadataServiceTemplate.Override.Service,
+			RouteOverrideSpec:   nil,
+		}
+
+		metadataSVCOverride, ctrlResult, err := metadataSD.CreateRouteAndServiceOverride(ctx, instance, helper)
+		if err != nil {
+			return ctrlResult, err
+		} else if (ctrlResult != ctrl.Result{}) {
+			return ctrlResult, nil
+		}
+
+		serviceDetails = append(
+			serviceDetails,
+			metadataSD,
+		)
+
+		if metadataServiceOverrides[cellName] == nil {
+			metadataServiceOverrides[cellName] = map[string]service.OverrideSpec{}
+		}
+
+		metadataServiceOverrides[cellName][string(service.EndpointInternal)] = *metadataSVCOverride
+		// metadata - end
+
+		// novncproxy
+		for _, endpointType := range []service.Endpoint{service.EndpointPublic, service.EndpointInternal} {
```

Comment:
The novncproxy is not created as a keystone endpoint so there is no concept of public and private novncproxy in nova. 

(This again shows that now you need to teach openstack-operator nova specific things)